From c4a974cc7f44bd6ebac7b1cdc90a90a65a1bd778 Mon Sep 17 00:00:00 2001
From: Giuseppe Scrivano <gscrivan@redhat.com>
Date: Wed, 5 Feb 2014 14:02:51 +0100
Subject: [PATCH] doc: remove SDL from virt-install documentation

and regenerate virt-install.pod, it includes also bits from a previous
commit: 'osdict: append "and later" to the last version of an OS'.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=886779

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>
---
 man/en/virt-install.pod    | 74 ++++++++++++++++++++++++----------------------
 man/en/virt-install.pod.in | 30 ++-----------------
 2 files changed, 42 insertions(+), 62 deletions(-)

diff --git a/man/en/virt-install.pod b/man/en/virt-install.pod
index 811b36c..520d26b 100644
--- a/man/en/virt-install.pod
+++ b/man/en/virt-install.pod
@@ -14,9 +14,9 @@ B<virt-install> is a command line tool for creating new KVM, Xen, or Linux
 container guests using the C<libvirt> hypervisor management library.
 See the EXAMPLES section at the end of this document to quickly get started.
 
-B<virt-install> tool supports both text based & graphical installations,
-using VNC or SDL graphics, or a text serial console. The guest can be
-configured to use one or more virtual disks, network interfaces,
+B<virt-install> tool supports graphical installations using (for example)
+VNC or SPICE, as well as text mode installs over serial console. The guest
+can be configured to use one or more virtual disks, network interfaces,
 audio devices, physical USB or PCI devices, among others.
 
 The installation media can be held locally or remotely on NFS, HTTP, FTP
@@ -325,7 +325,7 @@ Values for some recent OS options are:
 
 =over 2
 
-=item win7                 : Microsoft Windows 7
+=item win7                 : Microsoft Windows 7 and later
 
 =item vista                : Microsoft Windows Vista
 
@@ -333,21 +333,21 @@ Values for some recent OS options are:
 
 =item winxp                : Microsoft Windows XP
 
-=item win2k8               : Microsoft Windows Server 2008
+=item win2k8               : Microsoft Windows Server 2008 and later
 
 =item win2k3               : Microsoft Windows Server 2003
 
-=item freebsd8             : FreeBSD 8.x
+=item freebsd8             : FreeBSD 8.x and later
 
 =item generic              : Generic
 
-=item debianwheezy         : Debian Wheezy
+=item debianwheezy         : Debian Wheezy and later
 
 =item debiansqueeze        : Debian Squeeze
 
 =item debianlenny          : Debian Lenny
 
-=item fedora18             : Fedora 18
+=item fedora18             : Fedora 18 and later
 
 =item fedora17             : Fedora 17
 
@@ -365,15 +365,15 @@ Values for some recent OS options are:
 
 =item rhel4                : Red Hat Enterprise Linux 4
 
-=item sles11               : Suse Linux Enterprise Server 11
+=item sles11               : Suse Linux Enterprise Server 11 and later
 
 =item sles10               : Suse Linux Enterprise Server
 
-=item opensuse12           : openSuse 12
+=item opensuse12           : openSuse 12 and later
 
 =item opensuse11           : openSuse 11
 
-=item ubuntuquantal        : Ubuntu 12.10 (Quantal Quetzal)
+=item ubuntuquantal        : Ubuntu 12.10 (Quantal Quetzal) and later
 
 =item ubuntuprecise        : Ubuntu 12.04 LTS (Precise Pangolin)
 
@@ -735,12 +735,6 @@ actual VNC display allocated can be obtained using the C<vncdisplay>
 command to C<virsh> (or L<virt-viewer(1)> can be used which handles this
 detail for the use).
 
-sdl
-
-Setup a virtual console in the guest and display an SDL window in the
-host to render the output. If the SDL window is closed the guest may
-be unconditionally terminated.
-
 spice
 
 Export the guest's console using the Spice protocol. Spice allows advanced
@@ -818,10 +812,6 @@ This option is deprecated in favor of C<--graphics vnc,listen=LISTEN,...>
 
 This option is deprecated in favor of C<--graphics vnc,keymap=KEYMAP,...>
 
-=item  --sdl
-
-This option is deprecated in favor of C<--graphics sdl,...>
-
 =item  --nographics
 
 This option is deprecated in favor of C<--graphics none>
@@ -1137,8 +1127,36 @@ to the guest
 See C<http://libvirt.org/formatdomain.html#elementsSmartcard> for complete
 details.
 
+=item --redirdev=BUS[,OPTS]
+
+Add a redirected device.
+
+=over 4
+
+=item B<type>
+
+The redirection type, currently supported is B<tcp> or B<spicevmc>.
+
+=item B<server>
+
+The TCP server connection details, of the form 'server:port'.
+
+=back
+
+Examples of invocation:
+
+=over 4
+
+=item B<--redirdev usb,type=tcp,server=localhost:4000>
 
+Add a USB redirected device provided by the TCP server on 'localhost'
+port 4000.
 
+=item B<--redirdev usb,type=spicevmc>
+
+Add a USB device redirected via a dedicated Spice channel.
+
+=back
 
 =back
 
@@ -1246,20 +1264,6 @@ booting from PXE, using VNC server/viewer
        --graphics vnc \
        --os-variant fedora9
 
-Install a guest with a real partition, with the default QEMU hypervisor for
-a different architecture using SDL graphics, using a remote kernel and initrd
-pair:
-
-  # virt-install \
-       --connect qemu:///system \
-       --name demo \
-       --ram 500 \
-       --disk path=/dev/hdc \
-       --network bridge=eth1 \
-       --arch ppc64 \
-       --graphics sdl \
-       --location http://download.fedora.redhat.com/pub/fedora/linux/core/6/x86_64/os/
-
 Run a Live CD image under Xen fullyvirt, in diskless environment
 
   # virt-install \
diff --git a/man/en/virt-install.pod.in b/man/en/virt-install.pod.in
index 19037bc..e8e4a7f 100644
--- a/man/en/virt-install.pod.in
+++ b/man/en/virt-install.pod.in
@@ -14,9 +14,9 @@ B<virt-install> is a command line tool for creating new KVM, Xen, or Linux
 container guests using the C<libvirt> hypervisor management library.
 See the EXAMPLES section at the end of this document to quickly get started.
 
-B<virt-install> tool supports both text based & graphical installations,
-using VNC or SDL graphics, or a text serial console. The guest can be
-configured to use one or more virtual disks, network interfaces,
+B<virt-install> tool supports graphical installations using (for example)
+VNC or SPICE, as well as text mode installs over serial console. The guest
+can be configured to use one or more virtual disks, network interfaces,
 audio devices, physical USB or PCI devices, among others.
 
 The installation media can be held locally or remotely on NFS, HTTP, FTP
@@ -671,12 +671,6 @@ actual VNC display allocated can be obtained using the C<vncdisplay>
 command to C<virsh> (or L<virt-viewer(1)> can be used which handles this
 detail for the use).
 
-sdl
-
-Setup a virtual console in the guest and display an SDL window in the
-host to render the output. If the SDL window is closed the guest may
-be unconditionally terminated.
-
 spice
 
 Export the guest's console using the Spice protocol. Spice allows advanced
@@ -754,10 +748,6 @@ This option is deprecated in favor of C<--graphics vnc,listen=LISTEN,...>
 
 This option is deprecated in favor of C<--graphics vnc,keymap=KEYMAP,...>
 
-=item  --sdl
-
-This option is deprecated in favor of C<--graphics sdl,...>
-
 =item  --nographics
 
 This option is deprecated in favor of C<--graphics none>
@@ -1210,20 +1200,6 @@ booting from PXE, using VNC server/viewer
        --graphics vnc \
        --os-variant fedora9
 
-Install a guest with a real partition, with the default QEMU hypervisor for
-a different architecture using SDL graphics, using a remote kernel and initrd
-pair:
-
-  # virt-install \
-       --connect qemu:///system \
-       --name demo \
-       --ram 500 \
-       --disk path=/dev/hdc \
-       --network bridge=eth1 \
-       --arch ppc64 \
-       --graphics sdl \
-       --location http://download.fedora.redhat.com/pub/fedora/linux/core/6/x86_64/os/
-
 Run a Live CD image under Xen fullyvirt, in diskless environment
 
   # virt-install \
-- 
1.9.0

